uwrt_mars_rover_drivetrain/uwrt_mars_rover_drivetrain_description/config/costmap_parameters.yaml

```diff
@@ -0,0 +1,217 @@
+      trans_stopped_velocity: 0.25
+      short_circuit_trajectory_evaluation: True
+      stateful: True
+      critics: ["RotateToGoal", "Oscillation", "BaseObstacle", "GoalAlign", "PathAlign", "PathDist", "GoalDist"]
+      BaseObstacle.scale: 0.02
+      PathAlign.scale: 32.0
+      PathAlign.forward_point_distance: 0.1
+      GoalAlign.scale: 24.0
+      GoalAlign.forward_point_distance: 0.1
+      PathDist.scale: 32.0
+      GoalDist.scale: 24.0
+      RotateToGoal.scale: 32.0
+      RotateToGoal.slowing_factor: 5.0
+      RotateToGoal.lookahead_time: -1.0
+
+planner_server:
+  ros__parameters:
+    expected_planner_frequency: 20.0
+    planner_plugins: ["GridBased"]
+    GridBased:
+      plugin: "nav2_navfn_planner/NavfnPlanner"
+      tolerance: 0.5
+      use_astar: false
+      allow_unknown: true
+
+global_costmap:
+  global_costmap:
+    ros__parameters:
+      footprint_padding: 0.03
+      update_frequency: 1.0
+      publish_frequency: 1.0
+      global_frame: map
+      robot_base_frame: base_link
+      use_sim_time: True
+      robot_radius: 0.22 # should modify
+      resolution: 0.05
+      plugins: ["static_layer", "inflation_layer", "obstacle_layer"]
+      obstacle_layer:
+        plugin: "nav2_costmap_2d::ObstacleLayer"
+        enabled: True
+        observation_sources: scan pointcloud1 pointcloud2
+        footprint_clearing_enabled: true
+        max_obstacle_height: 2.0
+        combination_method: 1
+        scan:
+          topic: /lidar_controller/out
+          obstacle_max_range: 2.5
+          obstacle_min_range: 0.0
+          raytrace_max_range: 3.0
+          raytrace_min_range: 0.0
+          max_obstacle_height: 2.0
+          min_obstacle_height: 0.0
+          clearing: True
+          marking: True
+          data_type: "LaserScan"
+          inf_is_valid: false
+        pointcloud1: 
+          topic: /camera1_sensor/points
+          clearing: True
+          marking: True
+          data_type: "PointCloud2"
+        pointcloud2: 
+          topic: /camera2_sensor/points
+          clearing: True
+          marking: True
+          data_type: "PointCloud2"
+      static_layer:
+        plugin: "nav2_costmap_2d::StaticLayer"
+        map_subscribe_transient_local: True
+        enabled: true
+        subscribe_to_updates: true
+        transform_tolerance: 0.1
+      inflation_layer:
+        plugin: "nav2_costmap_2d::InflationLayer"
+        enabled: true
+        inflation_radius: 0.55
+        cost_scaling_factor: 1.0
+        inflate_unknown: false
+        inflate_around_unknown: true
+      always_send_full_costmap: True
+
+
+local_costmap:
+  local_costmap:
+    ros__parameters:
+      update_frequency: 5.0
+      publish_frequency: 100.0
+      global_frame: base_link
+      robot_base_frame: base_link
+      use_sim_time: True
+      rolling_window: true
+      width: 3
+      height: 3
+      resolution: 0.05
+
+      plugins: ["static_layer", "inflation_layer", "obstacle_layer"]
```

Comment:
Just took a peek at the setup guide any they are using voxel_layer and inflation_layer. 
If I understand correcltly, voxel layer is for 3D planning (which can be done with our stereocamera) and inflation layer is for avoidance, so good  to keep. Let''s get rid of static layer here. I'll let you decide between obstacle layer and voxel layer for the local planner. Seems like the difference between 2D/3D detections on the costmap to me. Maybe we start simple and keep obstalce layer so it's 2D? Will let you reply to this comment with your judgement

Reply:
I think for now it might be best to get a 2D costmap, so with a obstacle layer. Even if we have a voxel layer, since we are only planning for the drivetrain to move in 2D, it might be unnecessary. If we do end up needed it in the future, we can always just add the plugin!

Also just a note: we might need to make some custom costmap layers as well. If our camera has any noise in it, we'll probably need to filter it out (there is a DenoiseLayer, but it doesn't seem to be available for galactic, I'd have to look at it more later). 

And if there are hills in the scene, how would we differentiate between a climbable hill vs an obstacle.

Reply:
And we should have a static_layer, even if it is empty. The obstacle_layer doesn't work without a static_layer, it seems like a problem some other people have too: https://answers.ros.org/question/405789/local-costmaps-obstacle-layer-not-detecting-obstacles/